Modules/Microsoft365DSC/DSCResources/MSFT_EXOAuthenticationPolicy/MSFT_EXOAuthenticationPolicy.psm1

```diff
@@ -0,0 +1,587 @@
+
+        [Parameter()]
+        [System.String]
+        $CertificatePath,
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $CertificatePassword
+    )
+
+    Write-Verbose -Message "Setting Authentication Policy configuration for $Identity"
+
+    $currentAuthenticationPolicyConfig = Get-TargetResource @PSBoundParameters
+
+    #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
+
+    #region Telemetry
+    $ResourceName = $MyInvocation.MyCommand.ModuleName -replace "MSFT_", ""
+    $CommandName  = $MyInvocation.MyCommand
+    $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
+        -CommandName $CommandName `
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+    $ConnectionMode = New-M365DSCConnection -Workload 'ExchangeOnline' `
+        -InboundParameters $PSBoundParameters
+
+    $NewAuthenticationPolicyParams = @{
+        AllowBasicAuthActiveSync            = $AllowBasicAuthActiveSync
+        AllowBasicAuthAutodiscover          = $AllowBasicAuthAutodiscover
+        AllowBasicAuthImap                  = $AllowBasicAuthImap
+        AllowBasicAuthMapi                  = $AllowBasicAuthMapi
+        AllowBasicAuthOfflineAddressBook    = $AllowBasicAuthOfflineAddressBook
+        AllowBasicAuthOutlookService        = $AllowBasicAuthOutlookService
+        AllowBasicAuthPop                   = $AllowBasicAuthPop
+        AllowBasicAuthPowerShell            = $AllowBasicAuthPowerShell
+        AllowBasicAuthReportingWebServices  = $AllowBasicAuthReportingWebServices
+        AllowBasicAuthRpc                   = $AllowBasicAuthRpc
+        AllowBasicAuthSmtp                  = $AllowBasicAuthSmtp
+        AllowBasicAuthWebServices           = $AllowBasicAuthWebServices
+    }
+
+    # CASE: Authentication Policy doesn't exist but should;
+    if ($Ensure -eq "Present" -and $currentAuthenticationPolicyConfig.Ensure -eq "Absent")
+    {
+        Write-Verbose -Message "Authentication Policy '$($Identity)' does not exist but it should. Create and configure it."
+        New-AuthenticationPolicy -Name $Identity @NewAuthenticationPolicyParams
+    }
+    # CASE: Authentication Policy exists but it shouldn't;
+    elseif ($Ensure -eq "Absent" -and $currentAuthenticationPolicyConfig.Ensure -eq "Present")
+    {
+        Write-Verbose -Message "Authentication Policy '$($Identity)' exists but it shouldn't. Remove it."
+        Remove-AuthenticationPolicy -Identity $Identity -Confirm:$false
+    }
+    # CASE: Authentication Policy exists and it should, but has different values than the desired one
+    elseif ($Ensure -eq "Present" -and $currentAuthenticationPolicyConfig.Ensure -eq "Present")
+    {
+        Write-Verbose -Message "Authentication Policy '$($Identity)' exists. Updating settings."
+        Set-AuthenticationPolicy -Identity $Identity @NewAuthenticationPolicyParams
```

Comment:
Same as above, should be piped to | Null. I understand not all resource are doing it currently, but it's good practice to start doing it.

Reply:
Now piped to null